api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -88,4 +89,8 @@ message UdpProxyConfig {
   // The default if not specified is false, that means each data chunk is forwarded
   // to upstream host selected on first chunk receival for that "session" (identified by source IP/port and local IP/port).
   bool use_per_packet_load_balancing = 7;
+
+  // Configuration for :ref:`access logs <arch_overview_access_logs>`
+  // emitted by the this udp_proxy.
+  repeated config.accesslog.v3.AccessLog access_log = 8;
```

Comment:
@mattklein123 Hi, Matt! Is this okay to reuse config of tcp proxy or a simpler config like file path?

Reply:
> @mattklein123 Hi, Matt! Is this okay to reuse config of tcp proxy or a simpler config like file path?

@mattklein123 I find that access logger Instance class is designed to:
```
virtual void log(const Http::RequestHeaderMap* request_headers,
                   const Http::ResponseHeaderMap* response_headers,
                   const Http::ResponseTrailerMap* response_trailers,
                   const StreamInfo::StreamInfo& stream_info) PURE;
``` 
If udp access logging uses the same framework, the related base class may need to be refactored.

Reply:
@mattklein123 I find that access logger Instance class is designed to:

```
virtual void log(const Http::RequestHeaderMap* request_headers,
                   const Http::ResponseHeaderMap* response_headers,
                   const Http::ResponseTrailerMap* response_trailers,
                   const StreamInfo::StreamInfo& stream_info) PURE;
```
If udp access logging uses the same framework, the related base class may need to be refactored.

Reply:
I would recommend reusing as much config/code as possible, even if some fields don't have meaning. I would model this after the TCP proxy access logging as you already suggested.